editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/internal/ProjectTemplatesActivator.java

```diff
@@ -104,4 +146,84 @@ public ProjectTemplatesSharedImages(Bundle pluginBundle) {
 			addImage(IMAGE_FUSE_ICON, IMAGE_FUSE_ICON);
 		}
 	}
+
+	class POMChangedReporter implements IResourceChangeListener {
+		/*
+		 * (non-Javadoc)
+		 * @see org.eclipse.core.resources.IResourceChangeListener#resourceChanged(org.eclipse.core.resources.IResourceChangeEvent)
+		 */
+		@Override
+		public void resourceChanged(IResourceChangeEvent event) {
+			if (!listenForEvents.get()) return;
+			if (event.getType() == IResourceChangeEvent.POST_CHANGE) {
+				try {
+					event.getDelta().accept(new DeltaWalker());
+				} catch (CoreException ex) {
+					pluginLog().logError(ex);
+				}
+			}
+		}
+	}
+	
+	class DeltaWalker implements IResourceDeltaVisitor {
+		
+		private static final String MARKER_ID = ProjectTemplatesActivator.PLUGIN_ID + ".CamelCatalogVersionProblem";
+		
+		/*
+		 * (non-Javadoc)
+		 * @see org.eclipse.core.resources.IResourceDeltaVisitor#visit(org.eclipse.core.resources.IResourceDelta)
+		 */
+      	public boolean visit(IResourceDelta delta) {
+			IResource res = delta.getResource();
+			if (!isRelevantResource(res)) return true;
+			switch (delta.getKind()) {
+				case IResourceDelta.ADDED:
+				case IResourceDelta.CHANGED:
+					if (res.exists()) {
+						validateConfiguredCamelVersionMatchesLoadedCamelVersion(res);
+					}
+					break;
+				default: // nothing
+			}
+			return false; 
+      	}
+      	
+      	private boolean isRelevantResource(IResource res) {
+      		return 	res != null &&
+      				"pom.xml".equalsIgnoreCase(res.getName()) && 
+      				!"bin".equalsIgnoreCase(res.getParent().getName()) && 
+      				!"target".equalsIgnoreCase(res.getParent().getName());
```

Comment:
Why "bin" and "target" and not just pom.xml? Aren't we just interested in the version changes that start with the pom?